src/main/java/competition/subsystems/hood/HoodSubsystem.java

```diff
@@ -2,52 +2,85 @@
 import com.google.inject.Inject;
 import com.google.inject.Singleton;
 import competition.IdealElectricalContract;
-import xbot.common.command.BaseSubsystem;
+import xbot.common.command.BaseSetpointSubsystem;
+import xbot.common.command.XScheduler;
 import xbot.common.controls.actuators.XCANTalon;
 import xbot.common.injection.wpi_factories.CommonLibFactory;
 import xbot.common.math.MathUtils;
+import xbot.common.properties.BooleanProperty;
 import xbot.common.properties.DoubleProperty;
 import xbot.common.properties.PropertyFactory;
 
 @Singleton
-public class HoodSubsystem extends BaseSubsystem{
+public class HoodSubsystem extends BaseSetpointSubsystem{
 
+    private double calibrationOffset;
     final DoubleProperty extendPowerProp;
     final DoubleProperty retractPowerProp;
     final DoubleProperty maxAngleProp;
     final DoubleProperty minAngleProp;
+    final DoubleProperty defaultForwardHeadingProp;
+    final DoubleProperty ticksPerDegreeProp;
+    final BooleanProperty calibratedProp;
     final DoubleProperty currentAngleProp;
     private IdealElectricalContract contract;
-    double maxAngle;
-    double minAngle;
     public XCANTalon hoodMotor;
+    double goalAngle;
```

Comment:
you might want these to be DoubleProperty s instead so you can debug what's going on more easily?